### ISSUE TITLE
Silence missing-custom-element-compile-options

### ISSUE THREAD

**Issue by dnl-brnr** [sveltejs/eslint-plugin-svelte3#51]
Hi, can you please provide an example of how to configure an `eslintrc.js` to ignore the following warning?

<img width="941" alt="image" src="https://user-images.githubusercontent.com/58155720/71948845-73c3e300-319f-11ea-9075-53d276b6fc52.png">

From my interpretation of this repo's readme, I've tried the following without success:
```
module.exports = {
  parserOptions: {
    ecmaVersion: 2019,
    sourceType: 'module',
  },
  env: {
    es6: true,
    browser: true,
  },
  plugins: ['svelte3'],
  overrides: [
    {
      files: ['**/*.svelte'],
      processor: 'svelte3/svelte3',
    },
  ],
  rules: {
    // ...
  },
  settings: {
    'svelte3/compiler-options': {
      customElement: true,
    },
  },
}

```

Thanks!

**Comment by Conduitry** [comment]
That should work, and it's working for me. Do you have a small repo that could be cloned to demonstrate what you're seeing?

**Comment by dnl-brnr** [comment]
Setting up the requested repo and playing with it, I discovered that the issue was caused by **not** having the following configuration in VS Code's `settings.json`:

```
{
  "files.associations": {
    "*.svelte": "html"
  }
}
```

Thanks for your help, that solves my issue.

**Comment by chrisolsen** [comment]
This is not a VSCode specific issue. This issue should be re-opened.

**Comment by kyoshino** [comment]
This is still happening with Svelte 4 even if `customElement: true` is defined in `vite.config.js`. Please reopen.

- [our config](https://github.com/videomark/videomark/blob/master/vite.config.js)
- [our component](https://github.com/videomark/videomark/blob/master/src/lib/components/stats/stats-expander.svelte)

**Comment by Relaxe111** [comment]
I have same issue with svelte 4:
<img width="1116" alt="CleanShot 2023-07-30 at 09 25 36@2x" src="https://github.com/sveltejs/eslint-plugin-svelte3/assets/40892030/9744f213-ce71-42fd-acbf-675369c15c52">

**Comment by netronicus** [comment]
I'm having the same issue 
![image](https://github.com/sveltejs/eslint-plugin-svelte3/assets/12499556/438b229a-1bcd-466b-a6d0-46920a1557e5)

**Comment by willnationsdev** [comment]
I found that I can stop VS Code from reporting this in the Problems panel if you go to your settings at `Svelte > Plugin > Svelte > Compiler Warnings` (`svelte.plugin.svelte.compilerWarnings` in JSON) and add the key `missing-custom-element-compile-options` with the value `ignore`.

I often use the `extends` feature of the `customElement` key in `<svelte:options />` to define properties which I then need to define as an `export let` declaration in order to have its value reflected back to an attribute. This *also* generates a plethora of `unused-export-let` warnings in my web components, so you can likewise silence those using the above method as well.

Cheers!

**Comment by kyoshino** [comment]
I can confirm the inline error is gone with the VS Code setting, but the [`svelte-check`](https://github.com/sveltejs/language-tools/tree/master/packages/svelte-check) tool still gives the warning. [This blog post](https://nickpoulos.medium.com/laravel-svelte-direct-seamless-svelte-components-in-your-laravel-blade-templates-40f88e0e2fb1) says all that we need is a `svelte-ignore` comment, but this doesn’t work with Svelte 4.

```svelte
<!-- svelte-ignore missing-custom-element-compile-options -->
<svelte:options customElement="vm-stats" />
```

**Update**: `svelte-check --compiler-warnings "missing-custom-element-compile-options:ignore"` will silence the warning. Also, you can add the VS Code setting to `.vscode/settings.json` in your project. See my referenced commit below ⬇️
